### ISSUE TITLE
v6.0.1 Segmentation fault - Debian 9.x

### ISSUE THREAD

**Issue by meoso** [PowerShell/PowerShell#6287]
same as #4055 , #4361 ; already reviewed  #4320

Steps to reproduce
------------------

```shell
sudo apt remove --purge powershell
sudo apt install libc6 libgcc1 libgssapi-krb5-2 liblttng-ust0 libstdc++6 libcurl3 libunwind8 libuuid1 zlib1g libssl1.0.2 libicu57 libssl1.0.0 libssl1.0.2 libssl1.1 libicu52 libicu57 curl gnupg apt-transport-https
curl https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
sudo sh -c 'echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-debian-stretch-prod stretch main" > /etc/apt/sources.list.d/microsoft.list'
sudo apt update
sudo apt install -y powershell
pwsh
```

Expected behavior
-----------------

```none
expected normal working bahavior
```

Actual behavior
---------------

```none
after a few seconds at shell prompt, it will 'Segmentation fault'
```

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1
OS                             Linux 4.14.0-0.bpo.3-amd64 #1 SMP Debian 4.14.13-1~bpo9+1 (2018-01-14)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

AppImage version does not Segfault; however i cannot use it because unable to install PowerCLI in AppImage

**Comment by anmenaga** [comment]
@meoso to gather more context - in your case is it a clean installation of Debian9 or an upgrade?

**Comment by meoso** [comment]
new install, i will edit original post as per standard layout.

**Comment by MarcosCC** [comment]
I have the same issue.

Let me know if you want/need other details.

**Comment by anmenaga** [comment]
@MarcosCC can you please post output of `$PSVersionTable` ? Thanks.

**Comment by MarcosCC** [comment]
Here:

$PSVersionTable                                                                                                                                                                                   

Name                           Value                                                                                                                                                                      
----                           -----                                                                                                                                                                      
PSVersion                      6.0.1                                                                                                                                                                      
PSEdition                      Core                                                                                                                                                                       
GitCommitId                    v6.0.1                                                                                                                                                                     
OS                             Linux 4.15.0-1-amd64 #1 SMP Debian 4.15.4-1 (2018-02-18)                                                                                                                   
Platform                       Unix                                                                                                                                                                       
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                    
PSRemotingProtocolVersion      2.3                                                                                                                                                                        
SerializationVersion           1.1.0.1                                                                                                                                                                    
WSManStackVersion              3.0

**Comment by anmenaga** [comment]
@meoso @MarcosCC Can you please give an overview how you deploy your repro machines (e.g. where the image is taken from, what configuration changes are done, what is in /etc/apt/sources.list)?

I've tried on the following versions, but could Not repro the segfault... 
- Stable: Linux 4.9.0-4-amd64 #1 SMP Debian 4.9.65-3+deb9u1 (2017-12-23)
- Testing: Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14)
- Unstable: Linux 4.15.0-1-amd64 #1 SMP Debian 4.15.4-1 (2018-02-18)

**Comment by abaddon82** [comment]
Same issue. Debian 9 unstable, upgraded from 7->8->9.

```
Name                           Value                                                                                                                         
----                           -----                                                                                                                         
PSVersion                      6.0.1                                                                                                                         
PSEdition                      Core                                                                                                                          
GitCommitId                    v6.0.1                                                                                                                        
OS                             Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.13-1 (2018-01-14)                                                                     
Platform                       Unix                                                                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                       
PSRemotingProtocolVersion      2.3                                                                                                                           
SerializationVersion           1.1.0.1                                                                                                                       
WSManStackVersion              3.0
```


[gdb.txt](https://github.com/PowerShell/PowerShell/files/1799840/gdb.txt)

**Comment by meoso** [comment]
Debian 9 Stable+Backports. Upgraded from 7>8>9.

/etc/apt/sources.list
```
deb http://deb.debian.org/debian/ stable main contrib non-free
deb-src http://deb.debian.org/debian/ stable main contrib non-free

deb http://deb.debian.org/debian/ stable-updates main contrib non-free
deb-src http://deb.debian.org/debian/ stable-updates main contrib non-free

deb http://deb.debian.org/debian-security stable/updates main contrib non-free
deb-src http://deb.debian.org/debian-security stable/updates main contrib non-free

deb http://ftp.debian.org/debian stretch-backports main contrib non-free
deb-src http://ftp.debian.org/debian stretch-backports main contrib non-free

deb [arch=amd64] http://www.deb-multimedia.org stable contrib non-free
```

uname -a
`$ uname -a
Linux XXXX 4.14.0-0.bpo.3-amd64 #1 SMP Debian 4.14.13-1~bpo9+1 (2018-01-14) x86_64 GNU/Linux`

**Comment by MarcosCC** [comment]
Just got updated to 6.0.2:

$PSVersionTable                                                                                                                                                                           

Name                           Value                                                                                                                                                                      
----                           -----                                                                                                                                                                      
PSVersion                      6.0.2                                                                                                                                                                      
PSEdition                      Core                                                                                                                                                                       
GitCommitId                    v6.0.2                                                                                                                                                                     
OS                             Linux 4.15.0-1-amd64 #1 SMP Debian 4.15.4-1 (2018-02-18)                                                                                                                   
Platform                       Unix                                                                                                                                                                       
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                    
PSRemotingProtocolVersion      2.3                                                                                                                                                                        
SerializationVersion           1.1.0.1                                                                                                                                                                    
WSManStackVersion              3.0 

And got the segmentation fault again,

**Comment by meoso** [comment]
what can we provide to help this along?

**Comment by enekux** [comment]
Hi,
I am facing same issue here. In my case, I have a Debian 8 upgraded to 9:

PS /home/ea.admin> $PSVersionTable                                                                                                                                                                                                                                             

Name                           Value                                                                                                                                                                                                                                          
----                           -----                                                                                                                                                                                                                                          
PSVersion                      6.0.2                                                                                                                                                                                                                                          
PSEdition                      Core                                                                                                                                                                                                                                           
GitCommitId                    v6.0.2                                                                                                                                                                                                                                         
OS                             Linux 4.9.0-6-amd64 #1 SMP Debian 4.9.82-1+deb9u3 (2018-03-02)                                                                                                                                                                                 
Platform                       Unix                                                                                                                                                                                                                                           
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                        
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                            
SerializationVersion           1.1.0.1                                                                                                                                                                                                                                        
WSManStackVersion              3.0

**Comment by MarcosCC** [comment]
New update:
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.1
PSEdition                      Core
GitCommitId                    v6.1.0-preview.1
OS                             Linux 4.15.0-1-amd64 #1 SMP Debian 4.15.4-1 (2018-02-18)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

SegFault still happening.

**Comment by enekux** [comment]
Haha, I also confirm that the Update does not help!

How could we workaround it? I am in development face and I am in standby...

Regards,

**Comment by meoso** [comment]
```
PS> $PSVersionTable                                                                                                      

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.1
PSEdition                      Core
GitCommitId                    v6.1.0-preview.1
OS                             Linux 4.14.0-0.bpo.3-amd64 #1 SMP Debian 4.14.13-1~bpo9+1 (2018-01-14)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


PS > Segmentation fault
```

**Comment by anmenaga** [comment]
This seems to be a duplicate of #6190 .
In some situations .NET Core is loading wrong version of OpenSSL library.
As far as I can tell from discussion in that thread - [they've fixed it](https://github.com/dotnet/corefx/pull/27208), but PowerShell will be able to take advantage of the fix only after it is moved to v2.1 of .NET Core.
Somebody also reported a workaround by using symlinks - [here](https://github.com/dotnet/corefx/issues/24891).

**Comment by MarcosCC** [comment]
That's interesting @anmenaga 

I had installed libssl1.0.0 from a previous version of Debian for compatibility with an app that needed that version, one of the only 5 local packages I have installed in my system.

I removed it now... and Powershell does not segfault!

**Comment by enekux** [comment]
Nailed it!

Thank you guys for the feedback!

**Comment by cathode911** [comment]
I confirm, it works if uninstall old versions of libssl

**Comment by shanselman** [comment]
`sudo apt-get remove libssl1.0.0` works for me
